### PR TITLE
Add use-sorted-alphabetically.ts

### LINKED ISSUE #39: Port over `useSortAlphabetically` reducer hook
Port over the `useSortAlphabetically` reducer hook from one of the client projects. This issue depends on AndcultureCode/AndcultureCode.JavaScript.Core#62

### PR DESCRIPTION
**Pull request by mrjones2014**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)
Fixes #39

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -49,6 +49,14 @@ export { UnmatchedRoute } from "./interfaces/unmatched-route";
 
 // #endregion Interfaces
 
+// -----------------------------------------------------------------------------------------
+// #region Reducers
```
Comment (brandongregoryscott) [inline src/index.ts:53]: Hmm - while the core of this hook is a reducer, I wonder if we should keep it in the `hooks` folder & region for now until we start to have multiple reducer actors?

File: src/reducers/use-sorted-alphabetically.test.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import { render } from "@testing-library/react";
```
Comment (brandongregoryscott) [inline src/reducers/use-sorted-alphabetically.test.tsx:1]: Thanks for adding tests! 👏

File: src/reducers/use-sorted-alphabetically.test.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        const { getAllByTitle } = render(<TestApp />);
+        const items = getAllByTitle("item").map(
+            (el: HTMLElement) => el.innerHTML
+        );
+
+        // Assert
+        expect(items).toStrictEqual(sortedList);
+    });
+
+    test("When setValues is called with unsorted array, then values is set to a sorted array", () => {
```
Comment (brandongregoryscott) [inline src/reducers/use-sorted-alphabetically.test.tsx:41]: Nitpick - capitalization consistency

File: src/reducers/use-sorted-alphabetically.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    ReducerAction,
+    ReducerState,
+    useReducer,
+} from "react";
+
+// -----------------------------------------------------------------------------------------
+// #region Reducer
+// -----------------------------------------------------------------------------------------
+
+function useSortedAlphabetically<T>(
```
Comment (brandongregoryscott) [inline src/reducers/use-sorted-alphabetically.ts:14]: Mind adding a JSDoc comment above this?

**Review by brandongregoryscott — COMMENTED** [review]
Left some minor feedback/thoughts @mrjones2014 - thanks for porting this over though!

File: src/index.ts (hunk 1 above)
Comment (mrjones2014) [reply src/index.ts:53]: Now that I'm thinking about it, I think we could actually consider this a hook anyway. It's just a hook that uses a reducer.

**Comment by mrjones2014** [comment]
@brandongregoryscott pushed up some changes to address your feedback!

**Review by brandongregoryscott — APPROVED** [review]

**Pull request merged**